### PR TITLE
4.x : Fix inputs for create model usecase

### PR DESCRIPTION
**Pull request by deepakrkris**
- [ ] - Fix Inputs to create model, property, methods 
- [ ] - Fix to load workspace after copying files from templates
- [ ] - Fix model definition file names

### CODE REVIEW

File: connector/template-handler.js [hunk 1]
```diff
@@ -75,9 +105,12 @@ class TemplateHandler {
     });
   }
   static copyTemplateDir(dir, destinationPath, cb) {
-    ncp(dir, destinationPath, cb);
+    const options = {clobber: false};
+    fs.copy(dir, destinationPath, options, function(err) {
```
Comment (superkhau) [inline connector/template-handler.js]: `fs.copy(dir, destinationPath, options, cb);`

File: connector/workspace-handler.js [hunk 2]
```diff
@@ -39,16 +32,31 @@ class WorkspaceHandler {
         handler.loadMiddleware(workspace, filePath, erroredFiles, next);
       });
     });
-    function callback(err, results) {
+    return taskList;
+  }
+  static getFileList(workspace, cb) {
+    fsUtility.getConfigFiles(workspace.directory, function(err, files) {
```
Comment (superkhau) [inline connector/workspace-handler.js]: `fsUtility.getConfigFiles(workspace.directory, cb);`

File: component/datamodel/model.js [hunk 3]
```diff
@@ -89,9 +90,17 @@ class Model extends Entity {
   getFilePath() {
     const modelDef = this._content;
     const filePath = path.join(this._graph.directory, modelDef.facetName,
-      config.ModelDefaultDir, modelDef.name + '.json');
+      config.ModelDefaultDir, lodash.kebabCase(modelDef.name) + '.json');
     return filePath;
   }
+  getFacetName() {
+    const modelDef = this._content;
```
Comment (deepakrkris) [inline component/datamodel/model.js]: _content is from the graph base classes. It is in the Node class.

File: component/tasks.js [hunk 4]
```diff
@@ -77,10 +77,7 @@ class Tasks {
   refreshModel(modelId, cb) {
     const workspace = this;
     const model = workspace.getModel(modelId);
-    const parts = modelId.split('.');
-    const facetName = parts[0];
-    const modelName = parts[1];
-    fsUtility.readModel(facetName, modelName, workspace,
+    fsUtility.readModel(model.getFacetName(), model.getName(), workspace,
```
Comment (deepakrkris) [inline component/tasks.js]: agreed

File: component/tasks.js [hunk 5]
```diff
@@ -126,14 +123,16 @@ class Tasks {
...
-    if (facetName && modelName) {
-      const modelId = facetName + '.' + modelName;
-      workspace.refreshModel(modelId, cb);
+    const id = facetName + '.' + modelName;
+    if (!workspace.getModel(id)) {
+      new Model(workspace, id, fileData);
+      cb();
     } else {
-      cb(new Error('file ignored: ' + filePath));
+      cb('filepath ignored: ' + filePath);
```
Comment (deepakrkris) [inline component/tasks.js]: agreed

File: connector/template-handler.js [hunk 6]
```diff
@@ -1,12 +1,42 @@
...
+          TemplateHandler.copyTemplateDir(
+          dir,
+          workspace.getDirectory(),
+          next);
+        });
+      });
+    }
+    function afterCopy(err) {
+      if (err) return callback(err);
+      TemplateHandler.loadWorkspace(workspace, template, function(err) {
```
Comment (deepakrkris) [inline connector/template-handler.js]: agreed

File: connector/template-handler.js (hunk 1 above)
Comment (deepakrkris) [inline connector/template-handler.js]: agreed

File: connector/workspace-handler.js (hunk 2 above)
Comment (deepakrkris) [inline connector/workspace-handler.js]: agreed

File: component/tasks.js [hunk 7]
```diff
@@ -77,20 +77,20 @@ class Tasks {
...
+      workspace,
+      function(err, modelDef) {
+        if (err) return cb(err);
+        if (model) {
+          model.updateDefinition(modelDef);
+          cb(null, model.getDefinition());
+        } else {
+          workspace.createModelDefinition(modelId, modelDef);
+          cb(null, model.getDefinition());
+        }
```
Comment (superkhau) [inline component/tasks.js]:
```
+        if (model) {
+          model.updateDefinition(modelDef);
+        } else {
+          workspace.createModelDefinition(modelId, modelDef);
+       
+        }
   cb(null, model.getDefinition());

File: component/tasks.js [hunk 8]
```diff
@@ -126,14 +126,16 @@ class Tasks {
...
-    if (facetName && modelName) {
-      const modelId = facetName + '.' + modelName;
-      workspace.refreshModel(modelId, cb);
+    const id = facetName + '.' + modelName;
+    if (!workspace.getModel(id)) {
+      new Model(workspace, id, fileData);
+      cb();
     } else {
-      cb(new Error('file ignored: ' + filePath));
+      cb(new Error('Model is already loaded'));
```
Comment (superkhau) [inline component/tasks.js]:
```
if (workspace.getModel(id))
  return cb(new Error('...))

new Model
cb();

File: connector/template-handler.js [hunk 9]
```diff
@@ -1,12 +1,42 @@
...
+          TemplateHandler.copyTemplateDir(
+          dir,
+          workspace.getDirectory(),
+          next);
+        });
+      });
+    }
+    const afterCopy = (function(err) {
+      if (err) return callback(err);
+      this.loadWorkspace(workspace, template, (function(err) {
```
Comment (superkhau) [inline connector/template-handler.js]: if (err) return callback(err);

File: connector/workspace-handler.js [hunk 10]
```diff
@@ -1,27 +1,20 @@
 'use strict';
+
 const fsUtility = require('../component/datamodel/util/file-utility');
+const path = require('path');
```
Comment (superkhau) [inline connector/workspace-handler.js]: del

File: connector/workspace-handler.js [hunk 11]
```diff
@@ -1,27 +1,20 @@
...
-      fsUtility.getConfigFiles(workspace.directory, function(err, files) {
-        if (err) return next(err);
-        modelFiles.concat(files['Models']);
-        modelConfigFiles.concat(files['ModelConfig']);
-        dataSourceFiles.concat(files['DataSources']);
-        middlewareFiles.concat(files['Middleware']);
-        next();
-      });
-    });
+    const modelFiles = [].concat(files.Models);
```
Comment (superkhau) [inline connector/workspace-handler.js]: const modelFiles = files.Models || [];

Comment (deepakrkris) [inline connector/workspace-handler.js]: agreed

File: connector/workspace-handler.js (hunk 10 above)
Comment (deepakrkris) [inline connector/workspace-handler.js]: agreed

**Pull request merged**